app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/agenda/AgendaViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.davidtomas.taskyapp.features.agenda.presentation.agenda
+
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.davidtomas.taskyapp.features.agenda.domain.useCase.GetAgendaUseCase
+import kotlinx.coroutines.launch
+
+class AgendaViewModel(
+    private val getAgendaUseCase: GetAgendaUseCase
+) : ViewModel() {
+
+    var state by mutableStateOf(AgendaState())
+        private set
+
+    init {
+        viewModelScope.launch {
+            getAgendaUseCase().fold(
+                onError = {},
+                onSuccess = {
+                    Log.d("Agenda", it.toString())
+                    state = state.copy(
+                        agendaItems = it
+                    )
+                }
+            )
```

Comment:
Here, you're loading the agenda items directly from the API and showing them. In offline-first apps, you should better stick to the single source of truth principle and just load them from the DB. The UI can just listen to the DB flow of agenda items. If you want to fetch them remotely, you can have a dedicated function for that which inserts them into the DB which then automatically triggers the DB flow for the UI